### DISCUSSION TITLE
Artisan defaults to a completed roast before I finish...

### DISCUSSION THREAD

**Discussion by Alexs-Cormorant600** [artisan-roaster-scope/artisan#989] [General]
Is this a common flaw?  It is really annoying to look over and find that the software has finished the roast without any input from me.  I then discard the useless data.  Help?!

**Comment by MAKOMO** [comment]
The software automatically marks the CHARGE and DROP events if the corresponding flags are turned on (default) in the Events dialog (menu Config >> Events, 1st tab). The roast is not finished and the event marks can be reset if needed.

The detection of CHARGE and DROP is based on the bean temperature curve shape as it is seen on drum roasters. If your bean temperature signal is very noisy, or if you are roasting on an air roaster this does not work well. In case you have ideas on how to improve this mechanism, you are well come to contribute!

Artisan can be configured to automatically stop recording via alarm rules.

**Comment by Alexs-Cormorant600** [comment]
Not sure that this is what is happening.

I get some successful roasts completed with Artisan and some that Artisan automatically thinks are done, before FC or have even dropped.

It just jumps to do I want to save the profile?

Is that different?

A

**Comment by roasterdave** [comment]
To help further we need to see an example profile where this happened and also your settings file.  

Please attach an example Artisan *.alog profile and your current Artisan settings file (as exported via menu Help >> Save Setings as *.aset) file.  Rename each file to add a .txt extension or zip the files before uploading. You can drag-and-drop the files onto your reply.  Note that you must come to this Discussions web site to attach them.  They will not work if sent by email.

**Reply by poundy** [reply]
I'll second what Roasterdave and Makomo have said here, this is not a "common flaw" and is likely something you have configured or are doing - describing your usual workflow will possibly help here. Also would help to explain how long you've been using Artisan, and whether this is a new experience or whether you've only ever seen this.

The workflow is very well controlled, you will only get that prompt about saving if you're trying to start a new roast, and you should only get that if you or an alarm have stopped the previous roast. I've had a very noisy roaster since the outset and my BBPs are probably not "normal" so I always have auto detect charge and drop OFF as I was getting incorrect marking of those points, but even when I had auto-drop on, it wouldn't ever turn the roast off and start a new one...
